### ISSUE TITLE
BaseLoss does not pass through **kwargs to loss func

### ISSUE THREAD

**Issue by bearpelican** [fastai/fastai2#322]
```
lf = CrossEntropyLossFlat(weight=torch.arange(10).float())
lf2 = nn.CrossEntropyLoss(weight=torch.arange(10).float())
lf.func.weight, lf2.weight
```
Output: 
```
(None,
 tensor([0., 1., 2., 3., 4., 5., 6., 7., 8., 9.]))
```

Problem seems to be parameter replacement funcs_kwargs:
https://github.com/fastai/fastcore/blob/af61499e09f1e1a22a080d1c95368572d2ac6256/fastcore/foundation.py#L162
As disabling that line fixes the issue.

**Comment by sgugger** [comment]
Indeed, func_kwargs is not compatible with additional kwargs. Removed it from `BaseLoss`.
